### ISSUE TITLE
Explicitly handle not receiving any weather data from the weather source

### ISSUE THREAD

**Issue by t-ober** [ie3-institute/PowerSystemDataModel#554]
Currently we get a division by zero error later down the road when not receiving any weather values from the weather Source. We should throw an exception before explicitly stating that there we did not find any weather data!

**Comment by staudtMarius** [comment]
I think this could be resolved with #999. After this PR an exception is thrown if we cannot read the csv weather or csv coordinate source correctly.

**Comment by sebastian-peter** [comment]
> I think this could be partially resolved with #999. After this PR an exception is thrown if we cannot read the csv weather or csv coordinate source correctly.

Yes, but we're still not checking if no data is returned at all - although this might be more the responsibility of SIMONA
